flexmeasures_client.py

```diff
@@ -88,7 +89,9 @@ def get_device_message(self, kwargs, **fnc_kwargs):
                 self.log("Device message cannot be retrieved. Any previous charging schedule will keep being followed.")
 
         schedule = res.json()
-        self.set_state("input_text.chargeschedule", state="ChargeScheduleAvailable", attributes=schedule)
+        self.log(f"Schedule {schedule}")
+        # To trigger state change we add the date to the state. State change is not triggered by attriibutes.
+        self.set_state("input_text.chargeschedule", state="ChargeScheduleAvailable" + datetime.now(tz=pytz.utc).isoformat(), attributes=schedule)
 
     def post_udi_event(self, *args, **fnc_kwargs):
```

Comment:
For clarity, we should update the function names:
- `post_udi_event` becomes `trigger_schedule`
- `get_device_message` becomes `get_schedule`